packages/main/src/Table.js

```diff
@@ -496,11 +497,18 @@ class Table extends UI5Element {
 	_handleMultiSelect(event) {
 		const row = this.getRowParent(event.target);
 		const previouslySelectedRows = this.rows.filter(item => item.selected);
+		const selectAllCheckbox = this.getDomRef().querySelector(".ui5-table-select-all-checkbox");
 
 		row.selected = !row.selected;
 
 		const selectedRows = this.rows.filter(item => item.selected);
 
+		if (selectedRows.length === this.rows.length) {
+			selectAllCheckbox.checked = true;
```

Comment:
Would be better if you do this in declarative way, e.g. to create a state (getter or private property) and use in the template:
<ui5-checkbox ?checked={{allRowsSelected}} class="ui5-table-select-all-checkbox">
Can you try if would work this way?